solr/core/src/java/org/apache/solr/servlet/ServletUtils.java

```diff
@@ -310,7 +310,11 @@ protected static Span buildSpan(Tracer tracer, HttpServletRequest request) {
             .asChildOf(tracer.extract(Format.Builtin.HTTP_HEADERS, new HttpServletCarrier(request)))
             .withTag(Tags.SPAN_KIND, Tags.SPAN_KIND_SERVER)
             .withTag(Tags.HTTP_METHOD, request.getMethod())
-            .withTag(Tags.HTTP_URL, request.getRequestURL().toString());
+            .withTag(Tags.HTTP_URL, request.getRequestURL().toString())
+            .withTag("net.host.name", request.getServerName())
+            .withTag("net.host.port", request.getServerPort())
+            .withTag("net.peer.name", request.getRemoteHost())
+            .withTag("net.peer.port", request.getRemotePort());
```

Comment:
Is https://www.javadoc.io/static/io.opentracing/opentracing-api/0.31.0/io/opentracing/tag/Tags.html#PEER_HOSTNAME the right constant to use?

Reply:
Hmmm missing the `net.` it looks like... https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/tag/Tags.java#L88

Reply:
I'm no expert here, but the opentelemetry semantic conventions seem to have evolved and differ from what OpenTracing used. I don't know the history here, but OT is now part of OTEL, so I interpreted that `peer.hostname` is the old OT tag name while `net.peer.name` is the new OTEL name. Thus there was no constant available. 

The `opentelemetry-semconv` dependency which we currently do not pull in, contains such constants in [SemanticAttributes.java](https://github.com/open-telemetry/opentelemetry-java/blob/main/semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java). A bit overkill to pull that in for four string constants though.

The question then is - since this is shared code between Jaeger and OTEL, and they both expect different tag names (the others, SPAN_KIND, HTTP_METHOD, HTTP_URL are the same), if we should tag both variants, or find some way to choose tags based on active module. Trick is that in `ServletUtils` class we only have a tracer instance and don't know the active module. Only way I can think of is to do `tracer instanceof io.opentelemetry.opentracingshim.TracerShim`.

Reply:
Hmm, I wonder if I'll skip these four tags in this PR, and just do the refGuide change to keep it contained.

What I initially planned was to add the solr host name in the trace, not the http request host name (which could be an alias, and is also part of the URL). It could be possible to add the `host.name` tag in a static way as part of the Otel configurator and avoid touching this common part for now.

Once we replace OT instrumentation with OTEL instrumentation, we can revisit what tags to set by default.